src/v/storage/disk_log_impl.cc

```diff
@@ -410,7 +410,7 @@ disk_log_impl::request_eviction_until_offset(model::offset max_offset) {
 }
 
 ss::future<> disk_log_impl::adjacent_merge_compact(
-  compaction_config cfg, std::optional<model::offset> new_start_offset) {
+  const compaction_config& cfg, std::optional<model::offset> new_start_offset) {
```

Comment:
changed `compaction_config` parameter signature to `const&`- this function is called [here](https://github.com/redpanda-data/redpanda/blob/7ee1946e94feef9a6a6649b977a4d8c82fad285a/src/v/storage/disk_log_impl.cc#L1233-L1234) from `do_compact()`, so the lifetimes should be safe

Reply:
Removed `const&`.